.github/workflows/ci.yml

```diff
@@ -72,6 +75,12 @@ jobs:
         # Fail if the output JSON is not as expected.
         diff results.json expected_results.json
 
+    - uses: actions/checkout@ec3a7ce113134d7a93b817d10a8272cb61118579 # 2.4.0
+      with:
+        repository: 'ynfle/diff'
+        ref: 'ae470e7702fd07a6ea850b2a6f2de701130fa048'
+        path: 'diff'
+
```

Comment:
Non-blocking suggestion: I think we could download the `diff` package only once, and have a single source of truth for its version.

We could either:
1. Keep only the `actions/checkout` one, and download to a location that can be found when compiling `trunner.nim` above, and then `docker cp` from that path
2. Use `nimble install` to install, then `docker cp` from the path that Nimble installs to (e.g. using `nimble path diff` to get the path in a portable way).

Reply:
It's 2 different jobs. Is there a way to cache checkout?

The reason for 2 different approaches was because `nimble` is anyways on the first image and it will be in nim's path when compiling so it makes life easy. On the second image, it's easier to have it in a subdirectory and copy then asking nimble where it is.

Reply:
Ah. I just forgot that we have separate jobs (even though I'm the one who suggested that...). 

It's fine then. And I agree with the two different approaches, for our current setup.

I'd have a slight preference for defining the diff ref only once as a workflow-wide environmental variable inside `ci.yml`, but it's of no real importance.

Reply:
I agree that it's better.

Is there a way to do create customer "variables"?

Would you like to PR this and the other recommended change? I can if not.

Reply:
Nvm https://docs.github.com/en/actions/learn-github-actions/expressions#literals